3.11/aql/execution-and-performance-optimizer.md

```diff
@@ -717,6 +721,36 @@ The following optimizer rules may appear in the `rules` attribute of
   case, no AQL is executed on the DB-Servers. Instead, the Coordinator
   directly works with the documents on the DB-Servers.
 
+- `optimize-cluster-multiple-document-operations`:
+  Bulk `INSERT` operations in cluster deployments can be optimized to avoid
+  unnecessary overhead that AQL queries typically require for the setup and
+  shutdown in clusters.
+
+  This improvement also decreases the number of HTTP requests to the DB-Servers.
+
+  The following patterns are recognized:
+
+  - `FOR doc IN @docs INSERT doc INTO collection`, where `@docs` is a
+    bind parameter with an array of documents to be inserted
+  - `FOR doc IN [ { … }, { … }, … ] INSERT doc INTO collection`, where the `FOR`
+    loop iterates over an array of input documents known at query compile time
+  - `LET docs = [ { … }, { … }, … ] FOR doc IN docs INSERT doc INTO collection`,
+    where the `docs` variable is a static array of input documents known at
+    query compile time
+
+  If a query has such a pattern, and all of the following restrictions are met,
+  then the optimization is triggered:
+
+  - There are no following `RETURN` nodes (including any `RETURN OLD` or `RETURN NEW`)
+  - The `FOR` loop is not contained in another outer `FOR` loop or subquery
+  - There are no other operations (e.g. `LET`, `FILTER`) between `FOR` and `INSERT`
+  - `INSERT` is not used on a SmartGraph edge collection
+  - The `FOR` loop iterates over a constant, deterministic expression
+
+  The optimization then adds a `MultipleRemoteExecutionNode` to the query
+  execution plan, which takes care of inserting all documents into the collection
+  in one go. Further optimizer rules are skipped if the optimization is triggered.
```

Comment:
```suggestion
```suggestion
  The optimization then replaces the `InsertNode` and `EnumerateListNode` with a `MultipleRemoteExecutionNode` in the query
  execution plan, which takes care of inserting all documents into the collection
  in one go. Further optimizer rules are skipped if the optimization is triggered.
```